abyfy_errors.go

```diff
@@ -0,0 +1,104 @@
+package httpapi
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+type prePostWrappedNotFoundError struct {
+	pre  string
+	post string
+
+	wrapped ipld.ErrNotFound
+}
+
+func (e prePostWrappedNotFoundError) String() string {
+	return e.Error()
+}
+
+func (e prePostWrappedNotFoundError) Error() string {
+	return e.pre + e.wrapped.Error() + e.post
+}
+
+func (e prePostWrappedNotFoundError) Unwrap() error {
+	return e.wrapped
+}
+
+func abyfyIpldNotFoundFallbackToMSG(msg string) error {
```

Comment:
What does "abyfy" mean?

Reply:
It's the contraction of ABI-ify. ABI is for Application Binary Interface and is how I reference type & reflection based errors VS string based errors.

Reply:
Ah ok, that's confusing. What if we call these `errors.go`, `parseIPLDNotFound()`, `parseIPLDNotFoundWithFallbackMsg()`, and `parseIPLDNotFoundWithFallbackErr`?

Reply:
Done